barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -38,8 +38,39 @@ using namespace bb;
 std::filesystem::path avm_dump_trace_path;
 
 namespace bb::avm_trace {
+
+std::string to_name(TxExecutionPhase phase)
+{
+    switch (phase) {
+    case TxExecutionPhase::SETUP:
+        return "SETUP";
+    case TxExecutionPhase::APP_LOGIC:
+        return "APP_LOGIC";
+    case TxExecutionPhase::TEARDOWN:
+        return "TEARDOWN";
+    default:
+        throw std::runtime_error("Invalid tx phase");
+        break;
+    }
+}
+
+/**************************************************************************************************
+ *                              HELPERS IN ANONYMOUS NAMESPACE
+ **************************************************************************************************/
 namespace {
 
+template <size_t N>
+std::vector<PublicCallRequest> non_empty_call_requests(std::array<PublicCallRequest, N> call_requests_array)
```

Comment:
call_requests_array could be made a const reference I think.
Maybe using copy_if and a back_inserter for filling the target vector would be more elegant.